usaspending_api/common/pagination.py

```diff
@@ -1,40 +1,117 @@
 from collections import OrderedDict
 
 from rest_framework.response import Response
-from rest_framework.pagination import PageNumberPagination
+from rest_framework.pagination import BasePagination
+from rest_framework.settings import api_settings
+from django.template import loader
+from rest_framework.utils.urls import remove_query_param, replace_query_param
+from rest_framework.compat import template_render
 
 
-class UsaspendingPagination(PageNumberPagination):
-    # the default number of items to be returned on a single page
+class UsaspendingPagination(BasePagination):
+    # The default page size
     page_size = 100
-    # the name of the query param that can be used to override the
-    # default page size
-    page_size_query_param = 'limit'
-    # the param below sets the max number of items that a request will
-    # return, regardless of the page size specified in the request
+
+    # The maximum page size
     max_page_size = 500
 
+    # Page size query param
+    page_size_query_param = 'limit'
+
+    # Page query param
+    page_query_param = 'page'
+
+    # Use a lazy template (i.e. doesn't need to know the total number of pages)
+    template = 'rest_framework/pagination/previous_and_next.html'
+
+    def paginate_queryset(self, queryset, request, view=None):
+        self.request = request
+        self.limit = self.get_limit(request)
+        self.page = self.get_page(request)
+        self.offset = self.get_offset(request)
+        self.has_next_page = self.next_page_exists(queryset)
+
+        # Turn on the controls if we have multiple pages
+        if self.next_page_exists and self.template is not None:
+            self.display_page_controls = True
+
+        return list(queryset[self.offset:self.offset+self.limit])
+
+    def next_page_exists(self, queryset):
+        # If our next page has a count > 0, return true
+        return (bool(queryset[(self.offset+self.limit):(self.offset+self.limit*2)]))
+
+    def get_limit(self, request):
+        # Check both POST and GET parameters for limit
+        request_parameters = {**request.data, **request.query_params}
+
+        if 'limit' in request_parameters:
+            # We need to check if this is a list due to the fact we support both POST and GET
+            l = request_parameters['limit']
```

Comment:
I'm curious--why would `limit` be an integer in one type of request and a list in another type of request?